### ISSUE TITLE
Unreachable code if exception is caught in decorator

### ISSUE THREAD

**Issue by qi55wyqu** [pylint-dev/pylint#9754]
### Bug description

```python
import functools


class MyException(Exception):
    pass


def ignoreException(exception):
    def outerWrapper(func):
        @functools.wraps(func)
        def innerWrapper(*args, **kwargs):
            try:
                func(*args, **kwargs)
            except Exception as exc:  # pylint: disable=broad-exception-caught
                if not isinstance(exc, exception):
                    raise exc
        return innerWrapper
    return outerWrapper


@ignoreException(MyException)
def foo():
    raise MyException()
    return True


foo()
```

### Command used

```shell
pylint pylinttest.py
```

### Pylint output

```python
pylinttest.py:24:4: W0101: Unreachable code (unreachable)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.2.4
astroid 3.2.2
Python 3.10.6 (remotes/origin/squish-3.10.6-dirty:9facf30, Aug 18 2022, 04:38:35) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

### OS / Environment

Debian GNU/Linux 12

**Comment by raphaelreme** [comment]
In your example, `return True` is truly unreachable right ? Even if the exception is caught, the function will not start back where it was interrupted. You can replace `return True` by `print("Hello")` and see that it is indeed never called.

**Comment by Pierre-Sassoulas** [comment]
Thank you @raphaelreme, I think you're right.

**Comment by qi55wyqu** [comment]
You're right! My mistake, sorry to bother you
